default.nix

```diff
@@ -787,6 +787,83 @@ rec {
       '';
   };
 
+  # Helper function to filter tests by type
+  filter_tests = type: tests:
+    let
+      # Get all test names that match the pattern
+      debug_tests = builtins.filter (name: 
+        builtins.match ".*-dbg$" name != null || 
+        builtins.match ".*-debug$" name != null
+      ) (builtins.attrNames tests);
+      
+      # Get all test names that don't match the pattern
+      release_tests = builtins.filter (name:
+        builtins.match ".*-dbg$" name == null && 
+        builtins.match ".*-debug$" name == null
+      ) (builtins.attrNames tests);
+      
+      # Select which set of names to use
+      selected_names = if type == "debug" then debug_tests else release_tests;
+    in
+      # Get the actual derivations for the selected names
+      builtins.map (name: tests.${name}) selected_names;
+
+  # Release version - excludes debug tests
+  release-systems-go = nixpkgs.releaseTools.aggregate {
+    name = "release-systems-go";
+    constituents = [
+      moc
+      mo-ide
+      mo-doc
+      didc
+      deser
+      samples
+      rts
+      base-src
+      base-tests
+      base-doc
+      docs
+      report-site
+      shell
+      check-formatting
+      check-rts-formatting
+      check-generated
+      check-grammar
+      check-error-codes
+    ] ++
+    filter_tests "release" tests  # Only include release tests.
+    ++ builtins.attrValues js
+    ;
+  };
+
+  # Debug version - only includes debug tests
+  debug-systems-go = nixpkgs.releaseTools.aggregate {
+    name = "debug-systems-go";
+    constituents = [
+      moc
+      mo-ide
+      mo-doc
+      didc
+      deser
+      samples
+      rts
+      base-src
+      base-tests
+      base-doc
+      docs
+      report-site
+      shell
+      check-formatting
+      check-rts-formatting
+      check-generated
+      check-grammar
+      check-error-codes
+    ] ++
+    filter_tests "debug" tests  # Only include debug tests.
+    ++ builtins.attrValues js
+    ;
+  };
+
   all-systems-go = nixpkgs.releaseTools.aggregate {
```

Comment:
we probably don't need this one any more

Reply:
yup, deleted now.